tests/functional/blob/test_mover.py

```diff
@@ -118,3 +121,25 @@ def test_move_old_chunk(self):
         del chunk_headers['oio_version']
         del new_chunk_headers['oio_version']
         self.assertEqual(chunk_headers, new_chunk_headers)
+
+    def test_move_with_wrong_size(self):
+        if not self.chunk_method.startswith('ec'):
+            self.skipTest('Only works with EC')
+
+        orig_chunk = random.choice(self.chunks)
+        chunk_volume = orig_chunk['url'].split('/')[2]
+        chunk_id = orig_chunk['url'].split('/')[3]
+
+        mover = BlobMoverWorker(self.conf, None,
+                                self.rawx_volumes[chunk_volume])
+        meta, stream = mover.blob_client.chunk_get(orig_chunk['url'])
+        data = stream.read()
+        stream.close()
+        data = data[:-1]
```

Comment:
If someday you have to do that in a big stream, you can use `itertools.islice()`.